newIDE/app/src/AssetStore/AssetPackDialog.js

```diff
@@ -0,0 +1,210 @@
+// @flow
+import * as React from 'react';
+import {
+  type AssetShortHeader,
+  type AssetPack,
+} from '../Utils/GDevelopServices/Asset';
+import Text from '../UI/Text';
+import { t, Trans } from '@lingui/macro';
+import Dialog from '../UI/Dialog';
+import FlatButton from '../UI/FlatButton';
+import RaisedButton from '../UI/RaisedButton';
+import RaisedButtonWithSplitMenu from '../UI/RaisedButtonWithSplitMenu';
+import { Column, Line } from '../UI/Grid';
+import { LinearProgress } from '@material-ui/core';
+import { installAsset } from './InstallAsset';
+import EventsFunctionsExtensionsContext from '../EventsFunctionsExtensionsLoader/EventsFunctionsExtensionsContext';
+import { useResourceFetcher } from '../ProjectsStorage/ResourceFetcher';
+import { showErrorBox } from '../UI/Messages/MessageBox';
+
+const styles = {
+  linearProgress: {
+    flex: 1,
+  },
+};
+
+type Props = {|
+  assetPack: AssetPack,
+  assetShortHeaders: Array<AssetShortHeader>,
+  addedAssetIds: Array<string>,
+  onClose: () => void,
+  onAssetPackAdded: () => void,
+  project: gdProject,
+  objectsContainer: gdObjectsContainer,
+  events: gdEventsList,
+  onObjectAddedFromAsset: (object: gdObject) => void,
+|};
+
+export const AssetPackDialog = ({
+  assetPack,
+  assetShortHeaders,
+  addedAssetIds,
+  onClose,
+  onAssetPackAdded,
+  project,
+  objectsContainer,
+  events,
+  onObjectAddedFromAsset,
+}: Props) => {
+  const missingAssetShortHeaders = assetShortHeaders.filter(
+    assetShortHeader => !addedAssetIds.includes(assetShortHeader.id)
+  );
+  const allAssetsInstalled = missingAssetShortHeaders.length === 0;
+  const noAssetsInstalled =
+    !allAssetsInstalled &&
+    missingAssetShortHeaders.length === assetShortHeaders.length;
+
+  const resourcesFetcher = useResourceFetcher();
+  const [
+    isAssetPackBeingInstalled,
+    setIsAssetPackBeingInstalled,
+  ] = React.useState<boolean>(false);
+
+  const eventsFunctionsExtensionsState = React.useContext(
+    EventsFunctionsExtensionsContext
+  );
+
+  const onInstallAssetPack = React.useCallback(
+    async (assetShortHeaders: Array<AssetShortHeader>) => {
+      if (!assetShortHeaders || !assetShortHeaders.length) return;
+      setIsAssetPackBeingInstalled(true);
+      try {
+        const installOutputs = await Promise.all(
+          assetShortHeaders.map(assetShortHeader =>
+            installAsset({
+              assetShortHeader,
+              eventsFunctionsExtensionsState,
+              project,
+              objectsContainer,
+              events,
+            })
+          )
+        );
+        installOutputs.forEach(installOutput => {
+          installOutput.createdObjects.forEach(object => {
+            onObjectAddedFromAsset(object);
+          });
+        });
+
+        await resourcesFetcher.ensureResourcesAreFetched(project);
+
+        setIsAssetPackBeingInstalled(false);
```

Comment:
Should you set it back to false in a `finally` block?

Reply:
I thought about that, but the line below closes the modal (in case of success, but not in case of failure) so it doesn't work :/